File: fork-runner/src/main/java/com/shazam/fork/injector/stat/StatServiceLoaderInjector.java
package com.shazam.fork.injector.stat;

import com.shazam.fork.stat.StatServiceLoader;

import static com.shazam.fork.injector.ConfigurationInjector.configuration;

public class StatServiceLoaderInjector {

    private static StatServiceLoader INSTANCE = null;

    public static synchronized StatServiceLoader statServiceLoader() {
        if (INSTANCE == null) {
            INSTANCE = new StatServiceLoader(configuration().getSortingStrategy().statistics.path);
        }
        return INSTANCE;
    }
}

File: fork-runner/src/main/java/com/shazam/fork/injector/stat/StatServiceLoaderInjector.kt
package com.shazam.fork.injector.stat

import com.shazam.fork.injector.ConfigurationInjector.configuration
import com.shazam.fork.stat.StatServiceLoader

object StatServiceLoaderInjector {

    private val INSTANCE by lazy {
        StatServiceLoader(configuration().sortingStrategy?.statistics?.path ?: "")
    }

    @JvmStatic
    fun statServiceLoader(): StatServiceLoader {
        return INSTANCE
    }
}
